operator/cmd/operator/server_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddServerFlags(t *testing.T) {
+	tests := []struct {
+		desc                    string
+		force                   bool
+		maxConcurrentReconciles int
+	}{
+		{
+			desc:                    "no concurrent reconcile",
+			force:                   false,
+			maxConcurrentReconciles: 1,
+		},
+		{
+			desc:                    "enable concurrent reconcile",
+			force:                   false,
+			maxConcurrentReconciles: 2,
+		},
+		{
+			desc:                    "no concurrent reconcile and force to proceed",
+			force:                   true,
+			maxConcurrentReconciles: 1,
+		},
+		{
+			desc:                    "enable concurrent reconcile and force to proceed",
+			force:                   true,
+			maxConcurrentReconciles: 2,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			cmd := &cobra.Command{}
+			args := &serverArgs{}
+
+			addServerFlags(cmd, args)
+
+			cmd.PersistentFlags().Set("force", fmt.Sprintf("%t", tt.force))
+			cmd.PersistentFlags().Set("max-concurrent-reconciles", fmt.Sprintf("%+v", tt.maxConcurrentReconciles))
```

Comment:
I don't understand what this is doing? We don't need to add unit tests to ensure cobra works; this doesn't seem to test any Istio logic at all

Reply:
I think it is validating that addServerFlags is configuring the cobra command with passed Args correctly. That is the only part it is validating. But the value of that is debatable.

Reply:
This patch is to provide regression tests for all serverCmd Flags, especially for newly added concurrent reconcile and original force. There are few tests available in Operator and we are trying to use this patch to kick off efforts to fill in the gap if make sense.

Reply:
Updated a version with function tests of cmd execution to generate right options for istio controller.